jsvg/build.gradle.kts

```diff
@@ -27,13 +28,24 @@ dependencies {
     jmh(testLibs.svgSalamander)
     jmh(testLibs.batik)
 }
+
 tasks {
 
     compileTestJava {
         sourceCompatibility = JavaVersion.VERSION_21.toString()
         targetCompatibility = JavaVersion.VERSION_21.toString()
     }
 
+    jar {
+        bundle {
+            bnd(
+                "Bundle-SymbolicName: com.github.weisj.jsvg2",
+                "-exportcontents: !*.impl.*,*",
```

Comment:
I only added the 2 for testing purposes. I guess `-exportcontents` relates to which classes are exposed as API? In this case not everything outside `impl` subpackages should be considered as being public. I have defined some exports in the `module-info.java` file. Though it is not up to date. I will get back to you with which packages at the moment are considered to be public.

Reply:
> I guess `-exportcontents` relates to which classes are exposed as API? 

Yes exactly.

> In this case not everything outside `impl` subpackages should be considered as being public. I have defined some exports in the `module-info.java` file. Though it is not up to date. I will get back to you with which packages at the moment are considered to be public.

It's no problem to update that list. Just let me know what should be in there.

Btw. you can also use bnd-tools to generate the JPMS module-info.class which ensures that OSGi and JPMS metadata are in sync:
- https://bnd.bndtools.org/instructions/jpms-module-info-options.html
- https://bnd.bndtools.org/instructions/jpms-module-info.html

If you are interested I can try to make that work too, hoping that my Gradle (non) skills don't hinder me from doing it.

Reply:
> It's no problem to update that list. Just let me know what should be in there.

I'll get back to you once I have updated it. If I don't get to it soon enough I'll just adjust it myself.

 > Btw. you can also use bnd-tools to generate the JPMS module-info.class which ensures that OSGi and JPMS metadata are in sync:
> 
> * [bnd.bndtools.org/instructions/jpms-module-info-options.html](https://bnd.bndtools.org/instructions/jpms-module-info-options.html)
> * [bnd.bndtools.org/instructions/jpms-module-info.html](https://bnd.bndtools.org/instructions/jpms-module-info.html)
> 
> If you are interested I can try to make that work too, hoping that my Gradle (non) skills don't hinder me from doing it.

Does this also ensure that code conforms to the generated module-info.java? It does sound intriguing not having to maintain two places, where these things are defined, but with the current setup the build will fail if the module-info.java is not respected.